src/Dfc.CourseDirectory.WebV2/Features/Venues/AddVenue/Address.cshtml

```diff
@@ -0,0 +1,46 @@
+﻿@model Dfc.CourseDirectory.WebV2.Features.Venues.AddVenue.Address.Command
+@{
+    Layout = "_V2LayoutProviderContext";
+    ViewContext.SetLayoutData(title: "Add location");
+}
+
+<div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+        <form asp-action="Address" asp-all-route-data="RouteValuesHelper.FromQueryString()">
+            <h1 class="govuk-heading-xl">
+                Add location
+            </h1>
+
+            <govuk-input asp-for="AddressLine1">
+                <govuk-input-label>Address line 1</govuk-input-label>
+            </govuk-input>
+
+            <govuk-input asp-for="AddressLine2">
+                <govuk-input-label>Address line 2</govuk-input-label>
+            </govuk-input>
+
+            <govuk-input asp-for="Town" input-class="govuk-input--width-20">
+                <govuk-input-label>Town or city</govuk-input-label>
+            </govuk-input>
+
+            <govuk-input asp-for="County" input-class="govuk-input--width-20">
+                <govuk-input-label>County</govuk-input-label>
+            </govuk-input>
+
+            <govuk-input asp-for="Postcode" input-class="govuk-input--width-10">
+                <govuk-input-label>Postcode</govuk-input-label>
+            </govuk-input>
+
+            <govuk-button type="submit" class="govuk-!-margin-right-3">Continue</govuk-button>
+
+            @if (ViewBag.FromPublishPage == true)
+            {
+                <govuk-button asp-action="CheckAndPublish" asp-all-route-data="@RouteValuesHelper.FromQueryString()" class="govuk-button--secondary">Cancel</govuk-button>
+            }
+            else
+            {
+                <govuk-button type="submit" asp-action="Cancel" asp-all-route-data="@RouteValuesHelper.FromQueryString()" formmethod="post" class="govuk-button--secondary">Cancel</govuk-button>
```

Comment:
Why is this a submit button?

Reply:
Because it's a `POST` (as it mutates state)

Reply:
Missed that, apologies.